qiskit/circuit/library/standard_gates/z.py

```diff
@@ -176,12 +176,18 @@ def inverse(self):
         """Return inverted CZ gate (itself)."""
         return CZGate()  # self-inverse
 
-    # def to_matrix(self):
-    #     """Return a numpy.array for the CZ gate."""
-    #     return numpy.array([[1, 0, 0, 0],
-    #                         [0, 1, 0, 0],
-    #                         [0, 0, 1, 0],
-    #                         [0, 0, 0, -1]], dtype=complex)
+    def to_matrix(self):
+        """Return a numpy.array for the CZ gate."""
+        if self.ctrl_state:
```

Comment:
this should be the case for all controlled gates, not just CZ?

Reply:
Yes

Reply:
I added a test to verify this.